scala-cli.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+set -eu
+
+# This script automatically download scala-cli from latest release and cache via coursier
+
+# https://gist.github.com/lukechilds/a83e1d7127b78fef38c2914c4ececc3c
+LATATEST_TAG=$(
+    curl --silent "https://api.github.com/repos/VirtusLab/scala-cli/releases/latest" | grep '"tag_name":' | sed 's/.*"\(v.*\)",/\1/'
+)
+
+if [ "$(expr substr $(uname -s) 1 5 2>/dev/null)" == "Linux" ]; then
+  SCALA_CLI_URL="https://github.com/VirtusLab/scala-cli/releases/download/$LATATEST_TAG/scala-cli-x86_64-pc-linux.gz"
+  CACHE_BASE="$HOME/.cache/coursier/v1"
+else
```

Comment:
Let's check for `Darwin` too. If it's neither Linux nor Darwin, either exit with an error message, or default to Linux with a warning.